hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncRequestFutureImpl.java

```diff
@@ -1140,7 +1140,8 @@ private String buildDetailedErrorMsg(String string, int index) {
   @Override
   public void waitUntilDone() throws InterruptedIOException {
     try {
-      waitUntilDone(Long.MAX_VALUE);
+      long cutoff = (EnvironmentEdgeManager.currentTime() + this.operationTimeout) * 1000L;
```

Comment:
What if operationTimeout here is negative(which means no timeout), or Long.MAX_VALUE?

Reply:
> What if operationTimeout here is negative(which means no timeout), or Long.MAX_VALUE?

Good point, exclude that cases seems better, will fix later, thanks.

Reply:
The type of operationTimeout is int, so seems no need to consider Long.MAX_VALUE case.